qupath-gui-fx/src/main/java/qupath/lib/gui/charts/Charts.java

```diff
@@ -834,6 +823,268 @@ public static ScatterchartBuilder scatterChart() {
+		 * @return this builder
+		 */
+		public BarChartBuilder series(String name, Collection<Data<String, Number>> data) {
+			if (data instanceof ObservableList)
+				series.add(new XYChart.Series<>(name, (ObservableList<Data<String, Number>>)data));
+			else
+				series.add(new XYChart.Series<>(name, FXCollections.observableArrayList(data)));
+			return this;
+		}
+
+		@Override
+		protected void updateChart(BarChart<String, Number> chart) {
+			super.updateChart(chart);
+			chart.getData().setAll(series);
+
+			// If we have a hierarchy, and pathClasses, make the plot live
+			for (var s : series) {
+				for (var d : s.getData()) {
+					var extra = d.getExtraValue();
+					var node = d.getNode();
+					if (extra instanceof PathClass && node != null) {
+						PathClass pathClass = (PathClass)extra;
+						Integer color = pathClass.getColor();
+						String style = String.format("-fx-background-color: rgb(%d,%d,%d,%.2f);",
+								ColorTools.red(color), ColorTools.green(color), ColorTools.blue(color), markerOpacity);
+						node.setStyle(style);
+						node.addEventHandler(MouseEvent.ANY, e -> {
+							if (e.getEventType() == MouseEvent.MOUSE_CLICKED)
+								tryToSelect(pathClass, e.isShiftDown(), e.getClickCount() == 2);
+							else if (e.getEventType() == MouseEvent.MOUSE_ENTERED)
+								node.setStyle(style + ";"
+										+ "-fx-effect: dropshadow(three-pass-box, rgba(0,0,0,0.5), 4, 0, 1, 1);");
+							else if (e.getEventType() == MouseEvent.MOUSE_EXITED)
+								node.setStyle(style);
+						});
+					}
+				}
+			}
+
+		}
+
+		/**
+		 * Plot two measurements against one another for the specified objects.
+		 * @name the name of the data series (useful if multiple series will be plot, otherwise may be null)
+		 * @param pathObjects the objects to plot
+		 * @return this builder
+		 */
+		public BarChartBuilder classes(Collection<? extends PathObject> pathObjects) {
+			xLabel("PathClass");
+			yLabel("Count");
+			this.pathObjects.addAll(pathObjects);
+			return series(
+					null,
+					pathObjects,
+					(PathObject p) -> p.getPathClass());
+		}
+
+		@Override
+		protected BarChart<String, Number> createNewChart(Axis<String> xAxis, Axis<Number> yAxis) {
+			return new BarChart<>(xAxis, yAxis);
+		}
+
+		/**
+		 * Try to select an object if possible (e.g. because a user clicked on it).
+		 * @param pathClass the object to select
+		 * @param addToSelection if true, add to an existing selection; if false, reset any current selection
+		 * @param centerObject if true, try to center it in a viewer (if possible)
+		 */
+		private void tryToSelect(PathClass pathClass, boolean addToSelection, boolean centerObject) {
+			PathObjectHierarchy hierarchy = null;
+			if (imageData != null)
+				hierarchy = imageData.getHierarchy();
+			else if (viewer != null)
+				hierarchy = viewer.getHierarchy();
+			if (hierarchy == null)
+				return;
+			var objects = pathObjects.stream().filter(p -> p.getPathClass().equals(pathClass)).toList();
```

Comment:
Need to consider that the `PathClass` can be null here.

Reply:
I think this is handled now